playground/src/components/state-machines-list/state-machines-list.spec.js

```diff
@@ -81,11 +98,87 @@ describe('list view', () => {
         canManageDeveloperSettings: true,
       },
     });
-    await waitForElement(() => getByText(/There are 2 objects in the cache/i));
-    fireEvent.click(getByText('sm-1'));
+    await waitForElement(() =>
+      rendered.getByText(/There are 2 objects in the cache/i)
+    );
+    fireEvent.click(rendered.getByText('sm-1'));
     await wait(() => {
-      expect(history.location.pathname).toBe('/my-project/state-machines/sm1');
+      expect(rendered.history.location.pathname).toBe(
+        '/my-project/state-machines/sm1'
+      );
+    });
+    await waitForElement(() => rendered.getByText(/sm-1/i));
+  });
+});
+
+describe('details view', () => {
+  let rendered;
+  beforeAll(() => {
+    applyUiKitMocks();
+  });
+  describe('when request is successful', () => {
+    it('should render data on page', async () => {
+      rendered = renderApp({
+        route: '/my-project/state-machines/sm1',
+        sdkMocks: [createStateMachinesDetailSdkMock()],
+        permissions: {
+          canViewDeveloperSettings: true,
+          canManageDeveloperSettings: true,
+        },
+      });
+      await waitForElement(() => rendered.getByText(/sm-1/i));
+    });
+    it('should retrigger request if id changes', async () => {
+      const defaultMock = createStateMachinesDetailSdkMock();
+      rendered = renderApp({
+        route: '/my-project/state-machines/sm1',
+        sdkMocks: [
+          defaultMock,
+          {
+            ...defaultMock,
+            action: {
+              ...defaultMock.action,
+              payload: {
+                ...defaultMock.action.payload,
+                options: {
+                  id: 'sm2',
+                },
+              },
+            },
+            response: {
+              ...defaultMock.response,
+              id: 'sm2',
+              key: 'sm-2',
+            },
+          },
+        ],
```

Comment:
Maybe we can keep the setup here a bit slimmer in that we have different mocks define above?